### PR TITLE
Fallback treatment for swaps

### LINKED ISSUE #5717: Support custom swap events in accounting
## Problem

Currently, custom swap events are ignored during accounting because:
1.  there's no default accounting setting to handle it
2. the accountant for the specified counterparty does not support it.

Discussion is here -> https://discord.com/channels/657906918408585217/683245846070165504/1080865282396004423

## Specification

- Create a default accounting setting to handle such cases/scenario(swaps, etc.)

### PR DESCRIPTION
**Pull request by nebolax**
Closes #5717

## Checklist

- [ ] The PR modified the frontend, and updated the [user guide](https://github.com/rotki/rotki/blob/develop/docs/usage_guide.rst) to reflect the changes.

### CODE REVIEW

File: rotkehlchen/accounting/structures/base.py [hunk 1]
```diff
@@ -289,10 +289,10 @@ def __str__(self) -> str:
     def get_timestamp_in_sec(self) -> Timestamp:
         return ts_ms_to_sec(self.timestamp)
 
-    def get_type_identifier(self) -> str:
+    def get_type_identifier(self, include_counterparty: bool = True) -> str:
         """A unique type identifier for known event types"""
```
Comment (prettyirrelevant) [inline rotkehlchen/accounting/structures/base.py:293]: Add a docstring to describe what the presence of `include_counterparty` introduces.

**Review by prettyirrelevant — COMMENTED** [review]
a comment

File: rotkehlchen/accounting/transactions.py [hunk 2]
```diff
@@ -60,11 +60,18 @@ def process(
...
-                f'During transaction accounting found transaction event {event} '
-                f'with no mapped event settings. Skipping...',
+            event_settings = self.tx_event_settings.get(
+                event.get_type_identifier(include_counterparty=False),
             )
-            return 1
+            if (
+                event_settings is None or
+                event_settings.accounting_treatment != TxAccountingTreatment.SWAP  # for swaps we fallback to default treatment  # noqa: E501
+            ):
```
Comment (LefterisJP) [inline rotkehlchen/accounting/transactions.py:69]: 1. Looks better this way
2. What do you mean exactly?

Don't you want to say something more like: "For swaps we we have a default treatment but for the rest we bail"

```suggestion
            if (    # for swaps we fallback to default treatment
                event_settings is None or
                event_settings.accounting_treatment != TxAccountingTreatment.SWAP
            ):
```

**Review by LefterisJP — COMMENTED** [review]
Small comment. And needs rebase.

File: docs/changelog.rst [hunk 3]
```diff
@@ -12,6 +12,7 @@ Changelog
 * :bug:`-` Ignoring forked assets ETC, BCH and BSV for accounting should now also remove any pre-fork references of them and completely omit them from the PnL report.
 * :bug:`-` Users with kraken accounts with old data that were never purged and repulled will no longer have missing events.
 * :bug:`-` PnL report will now correctly show progress bar percentage if user has connected but non-syncing exchanges.
+* :bug:`5717` Swaps will now be processed in accounting correctly no matter the counterparty set.
```
Comment (LefterisJP) [inline docs/changelog.rst:15]: Ah and one more comment on changelog. Remember user-facing.

```suggestion
* :bug:`5717` Swaps will now be processed in accounting correctly even if set manually.
```

File: rotkehlchen/accounting/transactions.py (hunk 2 above)
Comment (nebolax) [reply rotkehlchen/accounting/transactions.py:69]: Yes

**Review by LefterisJP — APPROVED** [review]
LGTM

**Pull request merged**